app/products/calls/connection/websocket_client.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {EventEmitter} from 'events';
+
+import {encode} from '@msgpack/msgpack/dist';
+
+import Calls from '@constants/calls';
+import DatabaseManager from '@database/manager';
+import {getCommonSystemValues} from '@queries/servers/system';
+
+export default class WebSocketClient extends EventEmitter {
+    private readonly serverUrl: string;
+    private readonly wsPath: string;
+    private ws: WebSocket | null = null;
+    private seqNo = 0;
+    private connID = '';
+    private eventPrefix = `custom_${Calls.PluginId}`;
+
+    constructor(serverUrl: string, wsPath: string) {
+        super();
+        this.serverUrl = serverUrl;
+        this.wsPath = wsPath;
+    }
+
+    async initialize() {
+        const database = DatabaseManager.serverDatabases[this.serverUrl]?.database;
+        if (!database) {
+            return;
+        }
+
+        const system = await getCommonSystemValues(database);
+        const connectionUrl = (system.config.WebsocketURL || this.serverUrl) + this.wsPath;
+
+        this.ws = new WebSocket(connectionUrl);
```

Comment:
Why us this not using the network library built by us instead of the built-in WS?